subsys/net/lib/conn_mgr/conn_mgr.c

```diff
@@ -0,0 +1,197 @@
+/*
+ * Copyright (c) 2019 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(conn_mgr, CONFIG_NET_CONNECTION_MANAGER_LOG_LEVEL);
+
+#include <init.h>
+#include <kernel.h>
+#include <errno.h>
+#include <net/net_core.h>
+#include <net/net_if.h>
+#include <net/net_mgmt.h>
+
+#include <conn_mgr.h>
+
+u16_t iface_states[CONN_MGR_IFACE_MAX];
+
+K_SEM_DEFINE(conn_mgr_lock, 1, UINT_MAX);
+
+static enum net_conn_mgr_state conn_mgr_iface_status(int iface_idx)
```

Comment:
It is a bit confusing to have `iface_idx` as during the review I constantly mixed it to network interface index which is actually not. The value is just index to iface_states[] array. Should we call it `iface_states_idx` to make it unambiguous?

Reply:
yes, or index